### PR TITLE
Backward compatibility fix for the Conversation class

### PR DESCRIPTION
**Pull request by Rocketknight1**
Improve our handling of the old `past_user_inputs` property a bit, and revert a hack in the tests because it now works properly. Should fix all the tests that were red!

cc @ydshieh

### CODE REVIEW

**Review by Dezzj — COMMENTED** [review]
Veiwed

File: src/transformers/pipelines/conversational.py [hunk 1]
```diff
@@ -163,7 +165,13 @@ def _user_messages(self):
...
         # This is a legacy property for backwards compatibility. It is recommended to just directly access
-        # conversation.messages instead.
+        # conversation.messages instead. The modern class does not care about which messages are "processed"
+        # or not.
+        if not self._user_messages:
+            return []
+        if self.messages[-1]["role"] != "user" or self._num_processed_user_inputs == len(self._user_messages):
+            return self._user_messages
+
         return self._user_messages[:-1]
```
Comment (ydshieh) [inline src/transformers/pipelines/conversational.py:175]: Could we have a bit of comments about why this conditions and their output values?

Comment (ydshieh) [reply src/transformers/pipelines/conversational.py:175]: I mean hard to understand why `self._user_messages` when `self.messages[-1]["role"] != "user"`

Comment (Rocketknight1) [reply src/transformers/pipelines/conversational.py:175]: Added a comment!

**Review by ydshieh — APPROVED** [review]

**Pull request merged**